libraries/AP_Scripting/applets/param-set.lua

```diff
@@ -0,0 +1,164 @@
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 7), 'could not add param table')
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+    assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', PARAM_TABLE_PREFIX .. name))
+    return Parameter(PARAM_TABLE_PREFIX .. name)
+end
+
+--[[
+  // @Param: PARAM_SET_ENABLE
+  // @DisplayName: Param Set enable
+  // @Description: Param Set enable
+  // @Values: 0:Disabled,1:Enabled
+  // @User: Standard
+--]]
+local PARAM_SET_ENABLE = bind_add_param("ENABLE", 1, 1)
+
+-- initialize MAVLink rx with buffer depth and number of rx message IDs to register
+mavlink:init(5, 1)
+
+-- register message id to receive
+mavlink:register_rx_msgid(PARAM_SET_ID)
+
+-- handle PARAM_SET message
+local parameters_which_can_be_set = {}
+parameters_which_can_be_set["MAV_OPTIONS"] = true
+parameters_which_can_be_set["PARAM_SET_ENABLE"] = true
+parameters_which_can_be_set["BATT_ARM_MAH"] = true
+parameters_which_can_be_set["BATT_ARM_VOLT"] = true
+parameters_which_can_be_set["BATT_CAPACITY"] = true
+parameters_which_can_be_set["BATT_CRT_MAH"] = true
+parameters_which_can_be_set["BATT_CRT_VOLT"] = true
+parameters_which_can_be_set["BATT_FS_CRT_ACT"] = true
+parameters_which_can_be_set["BATT_FS_LOW_ACT"] = true
+parameters_which_can_be_set["BATT_LOW_MAH"] = true
+parameters_which_can_be_set["BATT_LOW_VOLT"] = true
+parameters_which_can_be_set["BRD_OPTIONS"] = true
+parameters_which_can_be_set["COMPASS_USE3"] = true
+parameters_which_can_be_set["FENCE_ACTION"] = true
+parameters_which_can_be_set["FENCE_ALT_MAX"] = true
+parameters_which_can_be_set["FENCE_ENABLE"] = true
+parameters_which_can_be_set["FENCE_RADIUS"] = true
+parameters_which_can_be_set["FENCE_TYPE"] = true
+parameters_which_can_be_set["LIGHTS_ON"] = true
+parameters_which_can_be_set["LOG_BITMASK"] = true
+parameters_which_can_be_set["LOG_DISARMED"] = true
+parameters_which_can_be_set["LOG_FILE_DSRMROT"] = true
+parameters_which_can_be_set["RTL_ALT"] = true
+parameters_which_can_be_set["RTL_LOIT_TIME"] = true
+parameters_which_can_be_set["RTL_SPEED"] = true
+
+function should_set_parameter_id(param_id)
+    if parameters_which_can_be_set[param_id] == nil then
+        return false
+    end
+    return true
+end
+
+function handle_param_set(msg)
+    gcs:send_text(MAV_SEVERITY.WARNING, string.format("%s: param set received", TEXT_PREFIX_STR))
+
+    -- ignore PARAM_SETs not aimed at our sysid:
+    if msg.target_system ~= MAV_SYSID:get() then
+        return
+    end
+    -- our component ID is assumed to be 1:
+    if msg.target_component ~= 1 then
+        return
+    end
+    -- check for this specific ID:
+    if not should_set_parameter_id(msg.param_id) then
+        gcs:send_text(MAV_SEVERITY.WARNING, string.format("%s: param set denied (%s)", TEXT_PREFIX_STR, msg.param_id))
+        return
+    end
+    param:set(msg.param_id, msg.param_value)
```

Comment:
This is set only, eg not save. The existing MAVLink checks for changes and possibly also saves.